src/components/AssetIcon.tsx

```diff
@@ -118,14 +117,48 @@ export const AssetIcon: React.FC<AssetIconProps> = ({
 
   // For Soroban custom tokens, use the Soroban logo
   if (token.type === AssetTypeWithCustomToken.CUSTOM_TOKEN) {
+    const tokenIdentifier = getTokenIdentifier(token);
+    let icon = icons[tokenIdentifier];
+    let imageUrl = icon?.imageUrl || "";
+
+    if (!imageUrl) {
+      const matchingKey = Object.keys(icons).find((key) =>
+        key.startsWith(`${token.code}:`),
```

Comment:
I'd say we shouldn't do this, as on the stellar network multiple different tokens could have the same token code so if we only search by the token code we'll risk displaying a wrong image for a given asset. We should always search icons using both the `token code` and `asset issuer`, which is what the `tokenIdentifier` represents.